core/CMakeLists.txt

```diff
@@ -59,44 +59,36 @@ macro (nn_check_lib LIB SYM DEF)
 endmacro (nn_check_lib)
 
 set(gateway_c_sources
+    ${dynamic_library_c_file}
     ./src/message.c
     ./src/message_queue.c
-    ./src/internal/event_system.c
-    ./src/gateway_internal.c
-    ./src/gateway.c
-    ./src/gateway_createfromjson.c
-    ./src/broker.c
     ./src/module_loader.c
-    ${dynamic_library_c_file}
 )
 
-
 set(gateway_h_sources
     ./inc/message.h
-    ./inc/message_queue.h
-    ./inc/broker.h
     ./inc/module.h
     ./inc/module_access.h
+    ./inc/module_loader.h
+    ./inc/dynamic_library.h
+    ../deps/parson/parson.h
```

Comment:
I would go as far to say, if the gateway and the native module host consume the same `some_library.a` in their respective `.dll`'s, then I think this is a necessary abstraction point